lib/Ast.ml

```diff
@@ -1802,14 +1836,16 @@ end = struct
      |{ctx= _; ast= Td _}
      |{ ctx= Cl _
       ; ast=
-          ( Pld _ | Top | Tli _ | Pat _ | Mty _ | Mod _ | Sig _ | Str _
-          | Clf _ | Ctf _ | Rep | Mb _ | Md _ ) }
+          ( Pld _ | Top | Tli _ | Tyv _ | Kab _ | Pat _ | Mty _ | Mod _
+          | Sig _ | Str _ | Clf _ | Ctf _ | Rep | Mb _ | Md _ | Jkd _ ) }
      |{ ctx=
-          ( Pld _ | Top | Tli _ | Typ _ | Cty _ | Pat _ | Mty _ | Mod _
-          | Sig _ | Str _ | Clf _ | Ctf _ | Rep | Mb _ | Md _ )
+          ( Pld _ | Top | Tli _ | Typ _ | Tyv _ | Kab _ | Cty _ | Pat _
+          | Mty _ | Mod _ | Sig _ | Str _ | Clf _ | Ctf _ | Rep | Mb _ | Md _
+          | Jkd _ )
       ; ast=
-          ( Pld _ | Top | Tli _ | Pat _ | Exp _ | Cl _ | Mty _ | Mod _
-          | Sig _ | Str _ | Clf _ | Ctf _ | Rep | Mb _ | Md _ ) } ->
+          ( Pld _ | Top | Tli _ | Tyv _ | Kab _ | Pat _ | Exp _ | Cl _
+          | Mty _ | Mod _ | Sig _ | Str _ | Clf _ | Ctf _ | Rep | Mb _ | Md _
+          | Jkd _ ) } ->
```

Comment:
(Having read the rest of the code, I think the answer is their parenthesization needs are explicitly spelled out in `Fmt_ast.fmt_jkind`)

Reply:
Yes, it seems this function is for helping deal with the complicated precedence hierarchy in the expression language. I don't think we need to sneak information about the kind language into `Prec.t`. And yes, we do handle the parenthesization within the formatting logic. I think this is fine for now, because the parenthesization rules are pretty simple. If the kind language grows substantially, we could think about taking a different approach.